docs/fairness/time-aware-fairness.md

```diff
@@ -0,0 +1,159 @@
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+<!-- /toc -->
+
+## Summary
+
+This document aims to describe how the scheduler can make use of historical resource usage by queues and users in order to achieve fairness of resource allocation over time. The mechanism of storing and serving the necessary data is out of scope for this document and will be described in a separate design doc.  
+The scheduler will consider past resource allocations when dividing over-quota resources to queues in the DRF algorithm: queues that have used more resource-hours should be penalized compared to queues that have not used resources recently. This will effect the over-quota resource division only, meaning that deserved quota and queue priority will take precedence.  
+This will be an opt-in feature: if the admin does not configure time-aware-fairness, the scheduler will behave exactly the same as before.
+
+## Motivation
+
+Time-aware fairness is a desirable behavior for many use cases, as evident by the popularity of HPC systems like slurm, and by the demand for slurm-like schedulers in k8s. Implementing time-aware fairness in KAI will offer users a k8s native scheduler that can offer deserved quotas, for inference and interactive use cases, alongside an HPC-like experience for batch job queueing. See the [user stories](#user-stories) section below for detailed use cases.
+
+### Goals
+
+* Allow fair time-sharing of resources: prioritize queues that have not accessed resources recently over queues that did. Other than deserved quotas, resources should be fairly allocated over time: queues with identical weights and requests will receive similar allocation of resource-hours over a long enough period of time.
+* Allow burst access to resources: allow queues in busy clusters to access more resources than their weight for short periods of time.
+* Fallback to current behavior: if over-time feature is not configured, the scheduler will behave exactly like today, with classic DRF determining fair share.
+* Don't break current guarantees of deserved resources and over-quota priority. 
+
+### Non-Goals
+* Guarantee feature parity with any existing HPC system
+* In-depth design of the accounting service: this will be addressed in a separate document
+
+## Proposal
+
+Track usage data of resources by queues over time, with a configurable decay formula and reset period. Use this usage data in DRF calculations when allocating resources: per resource, 
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
```

Comment:
We should maybe have slightly more requirments here
I think another requirement here is that a user should be able to:  Not use this feature at all (opt-in feature) - and all scheduler behavior should stay the same